### ISSUE TITLE
Bug - Nav: flyout isn't accessible

### ISSUE THREAD

**Issue by thatblindgeye** [patternfly/patternfly-react#7781]
**Describe the problem**
Followup to #7772 

Currently the [Nav flyout example](https://www.patternfly.org/v4/components/navigation#flyout) shows a nav item that both is a link and contains a flyout menu. Due to how some assistive tech navigates pages, the only way to open the flyout menu may be to press Enter or Space on the nav item. The PR linked above would add that functionality in, but because links can be activated via mouse click or pressing Enter, and because VoiceOver's VO modifier + Space keyboard shortcut acts as a click event, the only way for users to open the flyout menu with VO is to press only Space (though that isn't exactly clear depending on the browser and may not be flawless).

As noted in the linked PR, two possible solutions include:

- We recommend that flyouts not be put on nav items that are also links that can be clicked, and update the flyout example accordingly. This would cause the flyout to behave more similarly to the drilldown example.
- We update the icon indicating there is a flyout menu to make it an interactive button, similar to the [W3C flyout button as toggle example](https://www.w3.org/WAI/tutorials/menus/flyout/#use-button-as-toggle).

Additionally, the difference in how to navigate the first level nav items (can only use Tab to navigate between them) and the flyout menus (can only use arrow keys) may not be immediately apparent to users navigating via only keyboard.

**How do you reproduce the problem?**
Go to either the example link above or into the [preview build nav flyout](https://patternfly-react-pr-7772.surge.sh/components/navigation/#flyout) from the linked PR, and try to navigate and operate the flyout nav example with VO.

**Expected behavior**
Nav flyouts can be clearly perceived and operated by assistive tech

**Is this issue blocking you?**
N/A

**Screenshots**
N/A

**What is your environment?**
N/A

**What is your product and what release date are you targeting?**
N/A

**Any other information?**
N/A

**Comment by mcarrano** [comment]
I don't know that there was ever a requirement that menu items that trigger a flyout menu also need to contain navigation links. I would opt that this functionality be removed in favor of making this work like the drilldown examples. I'm a little concerned that making this change might break someone, but to my knowledge Satellite is the only project using flyouts right now. I checked and they don't need this capability.

@nicolethoen @tlabaj how do you feel about removing the ability for a parent node to be both for opening a flyout and for navigation? Do you remember if there was a specific reason we implemented it that way or who did the implementation? It is still marked as beta.

**Comment by mcarrano** [comment]
@kmcfaul I believe you did the current flyout nav implementation. Can you take a look at this? Any reason we can't remove this capability and have the flyout work like a drilldown menu?

**Comment by mcarrano** [comment]
Response from @kmcfaul 
> I think there shouldn't be too much trouble to implement either suggestion on the react side. The first suggestion would be more of a design decision as the recommendation for how a flyout menu is laid out changes slightly (that flyout items shouldn't be links but a descriptor to what the flyout contains). The second suggestion also seems feasible dividing the link from the toggle element and updating where the flyout logic is placed (on the toggle opposed to the whole nav item). It may result in some shuffling of where css/aria props get placed but I'm not positive if that would be required.

Let's just go ahead with the suggestion to update the flyout to work more like a drill-down menu, then.

**Comment by RuiBrian** [comment]
I can be assigned to this!

**Comment by RuiBrian** [comment]
@mcarrano @kmcfaul For this issue, are we looking to either:

1. Simply update the example (remove links from the nav items with flyout) to show the way we recommend using a nav with flyout. Then, it would be up to the end user to follow our recommendations and they can still use nav with flyout and links if they want
2. Actually disable the ability for a flyout to be added on a nav item that also serve as links (and also update the example accordingly)

Discussed with @thatblindgeye about this a little already, but was also wondering if you guys had an opinion on which choice is better or more suitable.

**Comment by mcarrano** [comment]
I think we are leaning towards no. 2, but interested in what others think. @nicolethoen @tlabaj ?

**Comment by RuiBrian** [comment]
Makes sense, thanks for the input!

**Comment by RuiBrian** [comment]
@kmcfaul I’ve started working on this issue in [this](https://github.com/patternfly/patternfly-react/pull/8279) draft PR but came across some things that could use some clarification.

1. I made conditional props so that a `NavItem` cannot have both a `to` prop and a `flyout` prop (these changes can be viewed [here](https://github.com/patternfly/patternfly-react/pull/8279/files). This is implemented by following a pattern described in [this](https://www.benmvp.com/blog/conditional-react-props-typescript/) article, and basically makes TypeScript throw an error when both props are passed in. Is this the desired behavior or should we instead strive for something simpler like a `console.log` or `console.error` message when both props happen to be passed in?
2. I also ran into another problem with the above implementation as it seems that the ability to tab to a `NavItem` is tied to the `to` prop. Therefore, a flyout `NavItem` without a `to` prop can no longer be tabbed to. Additionally, @thatblindgeye noted that a flyout `NavItem` may be announced as a link by screen readers which may be confusing for users. One possible fix for these issues is to wrap the `NavItem` in a button component if `flyout` is passed (this would also hopefully fix some issues I’ve been running into with the `aria-expanded` prop. Would this be the best way to fix this or are there better alternatives to consider?

**Comment by kmcfaul** [comment]
1. We have used `console.error` before to indicate invalid prop combinations but I think the method you've implemented is also fine. I'm guessing that might stop a build compared to the console.error which wouldn't, but since they're conflicting props I don't see an issue with it. @tlabaj @nicolethoen Any opinions?
2. From what I can tell going back over the code, a user should still be able to set the tabIndex manually in the children (`renderClonedChild` method). I would test to make sure that is still the case because I don't think we want to lose the tab behavior. For the announcing you can try to assign a `role` to the anchor (maybe `menuitem`?) to tell the screenreader it's not a link (as it defaults to being a link role with an href). If that doesn't work out swapping to a button should be fine as long as the styling isn't changed.

**Comment by nicolethoen** [comment]
> 1. I'm guessing that might stop a build compared to the console.error which wouldn't, but since they're conflicting props I don't see an issue with it. @tlabaj @nicolethoen Any opinions?

It's a very cool approach I hadn't seen before. I'm interested in @tlabaj's thoughts.
